src/main/java/org/opensearch/neuralsearch/rest/RestNeuralSparseClearCacheHandler.java

```diff
@@ -0,0 +1,100 @@
+ */
+
+@Log4j2
+public class RestNeuralSparseClearCacheHandler extends BaseRestHandler {
+    private static final String INDEX = "index";
+    public static String NAME = "neural_sparse_clear_cache_action";
+    private final ClusterService clusterService;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+
+    public RestNeuralSparseClearCacheHandler(ClusterService clusterService, IndexNameExpressionResolver indexNameExpressionResolver) {
+        this.clusterService = clusterService;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+    }
+
+    /**
+     * @return name of Clear Cache API action
+     */
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * @return Immutable List of Clear Cache API endpoint
+     */
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+            new Route(RestRequest.Method.POST, String.format(Locale.ROOT, "%s/%s/{%s}", NeuralSearch.NEURAL_BASE_URI, "clear_cache", INDEX))
+        );
+    }
+
+    /**
+     * @param request RestRequest
+     * @param client NodeClient
+     * @return RestChannelConsumer
+     */
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        NeuralSparseClearCacheRequest clearCacheRequest = createClearCacheRequest(request);
+        log.info("[Neural Sparse] ClearCache started for the following indices: [{}]", String.join(",", clearCacheRequest.indices()));
+        return channel -> client.execute(NeuralSparseClearCacheAction.INSTANCE, clearCacheRequest, new RestToXContentListener<>(channel));
+    }
+
+    // Create a clear cache request by processing the rest request and validating the indices
+    private NeuralSparseClearCacheRequest createClearCacheRequest(RestRequest request) {
+        String[] indexNames = Strings.splitStringByCommaToArray(request.param("index"));
+        Index[] indices = indexNameExpressionResolver.concreteIndices(clusterService.state(), strictExpandOpen(), indexNames);
+        validateIndices(indices);
+
+        return new NeuralSparseClearCacheRequest(indexNames);
+    }
+
+    // Validate if the given indices are k-NN indices or not. If there are any invalid indices,
+    // the request is rejected and an exception is thrown.
+    private void validateIndices(Index[] indices) {
+        List<String> invalidIndexNames = Arrays.stream(indices)
+            .filter(index -> !"true".equals(clusterService.state().metadata().getIndexSafe(index).getSettings().get(SPARSE_INDEX)))
+            .map(Index::getName)
+            .collect(Collectors.toList());
+
+        if (!invalidIndexNames.isEmpty()) {
+            throw new NeuralSparseInvalidIndicesException(
+                invalidIndexNames,
+                "ClearCache request rejected. One or more indices have 'index.neural_sparse' set to false."
```

Comment:
What about:
1. Replace `One or more indices` with `Indices %s`
2. `have 'index.neural_sparse' set to false` -> don't support clear cache operation.